src/99-appendices/03-hed.md

```diff
@@ -66,44 +74,114 @@ Example:
+similar to categorical values, except that the HED string must contain exactly one `#` specifying a placeholder
+for the actual column values. Tools are responsible for substituting the actual column values for the `#` during
+analysis.
+
+Example: An `*_events.tsv` containing a categorical column (`trial_type`) and two value
+columns (`response_time` and `stim_file`).
+
+```Text
+onset  duration  trial_type  response_time stim_file
+1.2    0.6       go          1.435         images/red_square.jpg
+5.6    0.6       stop        1.739         images/blue_square.jpg
+```
+Example: An accompanying `*_events.json` sidecar describing both categorical and value columns.
+
+```JSON
+{
+   "trial_type": {
+       "LongName": "Event category",
+       "Description": "Indicator of type of action that is expected",
+       "Levels": {
+          "go": "A red square is displayed to indicate starting",
+          "stop": "A blue square is displayed to indicate stopping",
+       },
+       "HED": {
+          "go": "Event/Category/Experimental stimulus, Event/Label/RedSquare,
+          Event/Description/A red square is displayed to indicate starting,
+          Sensory presentation/Visual, Item/Object/2D Shape/Square,
+          Attribute/Visual/Color/Red, Attribute/Visual/Rendering type/Screen,
+          Attribute/Location/Screen/Center",
+          "stop": "Event/Category/Experimental stimulus, Event/Label/BlueSquare,
+          Event/Description/A blue square is displayed to indicate stopping,
+          Sensory presentation/Visual, Item/Object/2D Shape/Square,
+          Attribute/Visual/Color/Blue, Attribute/Visual/Rendering type/Screen,
+          Attribute/Location/Screen/Center",
+       }
+   },
+   "response_time": {
+       "LongName": "Response time after stimulus",
+       "Description": "Time from stimulus presentation until subject presses button",
+       "Units": "ms",
+       "HED": "Attribute/Response start delay/# ms, Action/Button press"
+   },
+   "stim_file": {
+       "LongName": "Stimulus file name",
+       "Description": "Relative path of the stimulus image file",
+       "HED": "Attribute/File/#"
+   }
+}
+```
+
+## Best practices
+
+Most studies will have event categorical variables and value variables that are common across many
+of the datasets in the study. You should try to annotate these columns in a `*_events.json` sidecar
+as high in the study hierarchy as possible to avoid duplicate annotations. Annotations that can be
+placed in sidecars are preferred to those placed directly in the HED column, because they are simpler,
+more compact, and less prone to inconsistent annotation. Downstream tools should not distinguish
+between tags specified using the explicit HED column and the categorical specifications, but should
+form the union before analysis. Further, the [inheritance principle](../02-common-principles.md#the-inheritance-principle)
+applies, so the data dictionaries can appear higher in the BIDS hierarchy.
```

Comment:
line breaks:

```suggestion
Most studies will have event categorical variables and value variables that are common across many
of the datasets in the study.
You should try to annotate these columns in a `*_events.json` sidecar
as high in the study hierarchy as possible to avoid duplicate annotations.
Annotations that can be placed in sidecars are preferred to those placed directly in the HED column,
because they are simpler, more compact, and less prone to inconsistent annotation.
Downstream tools should not distinguish between tags specified using the explicit HED column and
the categorical specifications, but should form the union before analysis.
Further, the [inheritance principle](../02-common-principles.md#the-inheritance-principle) applies,
so the data dictionaries can appear higher in the BIDS hierarchy.
```